webapp/src/components/common/DataGridForm.tsx

```diff
@@ -107,16 +107,49 @@ function DataGridForm<TData extends Data>({
 
   useEffect(() => setRef(apiRef, { data, setData, formState }), [apiRef, data, setData, formState]);
 
+  // Rows cannot be added or removed, so no dependencies are needed
+  const rowNames = useSafeMemo(() => Object.keys(defaultData), []);
+
+  const columnsWithAdjustedSize = useMemo(
+    () =>
+      columns.map((column) => ({
+        ...column,
+        width:
+          "width" in column
+            ? column.width
+            : Math.max(
+                measureTextWidth(column.title),
+                ...rowNames.map((rowName) =>
+                  measureTextWidth(defaultData[rowName][column.id].toString()),
+                ),
+              ),
+      })),
+    [columns, defaultData, rowNames],
+  );
```

Comment:
here a canvas is re-created at each iteration in the Math.max(), it's expensive

```ts
  const columnsWithAdjustedSize = useMemo(() => {
    const measureText = createTextMeasurer();

    return columns.map((column) => {
      if ("width" in column) {
        return column;
      }

      const maxWidth = Math.max(
        measureText(column.title),
        ...rowNames.map((rowName) => measureText(defaultData[rowName][column.id].toString())),
      );

      return { ...column, width: maxWidth };
    });
  }, [columns, defaultData, rowNames]);
  ```
  
  see comment about `domUtils.ts` file for the `createTextMeasurer()` util